distribution/lib/Standard/Base/0.0.0-dev/src/System/File/File_Permissions.enso

```diff
@@ -0,0 +1,103 @@
+from Standard.Base import all
+
+polyglot java import java.nio.file.attribute.PosixFilePermission
+
+type Permission
+    ## Permission for read access for a given entity.
+    type Read
+
+    ## Permission for write access for a given entity.
+    type Write
+
+    ## Permission for execute access for a given entity.
+    type Execute
+
+type File_Permissions
+    ## Access permissions for a file.
+    type File_Permissions (owner : Vector Permission) (group : Vector Permission) (others : Vector Permission)
+
+    ## Converts the Enso atom to its Java enum counterpart.
+    to_java : Vector PosixFilePermission
+    to_java =
+        result = Vector.new_builder
+        if self.owner.contains Read then
```

Comment:
Shouldn't this be more functional. Something like:
```
result = Vector.new_builder
result1 = if self.owner.contains Read then
  result.append PosixFilePermission.OWNER_READ
else
  result
result2 = if self.owner.contains Write then
  result1.append PosixFilePermission.OWNER_WRITE
else
  result1
```

Reply:
The `Vector_Builder` is a mutable helper, so it doesn't have to be reassigned on each change.

We could do
```
result = []
result1 = if self.owner.contains Read then result + [PosixFilePermission.OWNER_READ] else result
...
```
but that seems just unnecessarily more complex